agent/src/agent/modules/functions/transform.py

```diff
@@ -0,0 +1,39 @@
+import re
+
+from typing import Callable
+
+TO_UPPER = 'to_upper'
+RE_SUB = 'regex_substring'
+
+# this is a number of arguments for each function that must be provided in the pipeline config
+# probably it's not the best place for them, think how to make validation better
+NUM_FUNCTION_ARGS = {TO_UPPER: 0, RE_SUB: 2}
+
+
+def to_upper(s: str) -> str:
+    return s.upper()
+
+
+def regex_substring(string: str, pattern: str, repl: str) -> str:
+    res = re.sub(pattern, repl, string)
+    if res == 'no_match':
+        raise Exception(f'Pattern `{pattern}` didn\'t match anything for the string `{string}`, repl `{repl}`')
+    return res
+
+
+def get_by_name(name: str) -> Callable:
```

Comment:
you can use `getattr` to get the function from a module by it's name instead of if/else, all names match anyway. It will throw AttributeError if no such function. The same is for lookup also [https://stackoverflow.com/questions/3061/calling-a-function-of-a-module-by-using-its-name-a-string](url)